Settings Screen

sravank9 opened the pull request:
- Implemented Settings screen

https://user-images.githubusercontent.com/108011305/181016188-699f5c5c-968c-4e8c-9dc5-787783a925b1.mp4

sravank9 committed:
Implemented Settings screen

puliajay43 committed:
Merge branch 'main' into feature/settings

puliajay43 committed:
WIP: Integrating DashXClient with settings preferences

sravank9 committed:
Implemented Settings Screen

sravank9 committed:
Fixed toggle and identify issues

sravank9 committed:
Fixed Switch UI issue

sravank9 committed:
Refactored Code

RabidFire commented on app/src/main/java/com/dashxdemo/app/application/MyApplication.kt:
```diff
@@ -0,0 +1,11 @@
+class MyApplication: Application() {
+    override fun onCreate() {
+        super.onCreate()
+        DashXClient.createInstance(this,"TLy2w3kxf8ePXEyEjTepcPiq","https://api.dashx-staging.com/graphql","staging")
```
Keys shouldn't be hard-coded. We should provide best practices -- how the user would configure this (per environment) in their application.

RabidFire replied:
These should not be hard-coded.

RabidFire commented on app/src/main/java/com/dashxdemo/app/api/responses/SavedPreferencesResponse.kt:
```diff
@@ -0,0 +1,15 @@
+
+import com.google.gson.annotations.SerializedName
+
+data class SavedPreferencesResponse(
```
Remove this class

RabidFire commented on app/src/main/java/com/dashxdemo/app/api/responses/StoredPreferenceResponse.kt:
```diff
@@ -0,0 +1,23 @@
+
+import com.google.gson.annotations.SerializedName
+
+data class StoredPreferenceResponse(
```
Rename to StoredPreferencesResponse

RabidFire commented on app/build.gradle:
```diff
@@ -43,6 +43,9 @@ dependencies {
     implementation 'com.google.android.material:material:1.6.1'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
 
+    //DashX
```
Add space after `//`

RabidFire commented on app/src/main/java/com/dashxdemo/app/api/responses/StoredPreferenceResponse.kt:
```diff
@@ -0,0 +1,23 @@
+    val newPost: StoredPreferenceData
+)
+
+data class StoredPreferenceData(
```
We need to use the SDK class now

RabidFire commented on app/src/main/res/navigation/navigation_login.xml:
```diff
@@ -36,4 +36,5 @@
         android:id="@+id/nav_contact"
         android:name="com.dashxdemo.app.feature.login.fragment.ContactFragment"
         android:label="ContactFragment" />
+
```
Remove extra space

RabidFire commented on app/src/main/res/values/strings.xml:
```diff
@@ -32,6 +32,12 @@
 
+    <string name="whenCreateAPost">When someone creates a post</string>
+    <string name="createPostNotification">This will send a notification when someone creates a post</string>
+    <string name="whenBookmarksAPost">When someone bookmarks a post</string>
+    <string name="bookmarksPostNotification">"This will send a notification when someone bookmarks a post   "</string>
```
Why are there quotes and extra spaces here in this string?

RabidFire replied:
The string names for the preferences can be based on the preference identifiers themselves:

```
- newPostNotification
- newPostNotificationHint
- newBookmarkNotification
- newBookmarkNotificationHint
```

sravank9 committed:
Bookmarks screen (#5)

* Implemented Home screen (#2)

* Added BottomNavigationView and navigation in Home screen

- Added BottomNavigationView
- Added required fragments in Home screens
- Added Navigation between the fragments

* Fixed build issue in resources

* Added button styles and Refactored code

* Implemented profile screen.

* Update ApiClient.kt

Co-authored-by: sravan9 <sravan@tilicho.in>
Co-authored-by: Pradeep Kumar <pradeep@keepworks.com>

* Implemented posts screen.

* Refactored Posts screens.

- Added Create Post Feature.

* Implemented Bookmark/Unbookmark a Post feature

* Implemented Bookmarks flow

* Update fragment_bookmarks.xml

* Refactored code

* Updated Response type

* Updated Responses for Posts and Bookmarks

* Refactored code

* WIP: Fixing bookmark removal issue

* Fixed crash in bookmarks screen

* Modified PR changes

* Code refactoring

* Implemented P.R changes

* Updated P.R changes

Co-authored-by: puliajay43 <35003996+puliajay43@users.noreply.github.com>
Co-authored-by: Pradeep Kumar <pradeep@keepworks.com>

sravank9 committed:
Merge branch 'develop' into feature/settings

sravank9 committed:
Merge branch 'main' into feature/settings

sravank9 commented:
@RabidFire  We need to push our SDK changes, to get this code run
We can get back here once the SDK changes were pushed to maven

sravank9 committed:
WIP: Working on P.R changes

sravank9 committed:
Updated with SDK changes

RabidFire commented on app/build.gradle:
```diff
@@ -13,6 +13,13 @@ android {
         versionCode 1
         versionName '1.0.0'
 
+        Properties localProperties = new Properties()
+        localProperties.load(project.rootProject.file('local.properties').newDataInputStream())
+
+        buildConfigField 'String', 'DASHX_PUBLIC_KEY', localProperties.getProperty('DASHX_PUBLIC_KEY')
+        buildConfigField 'String', 'DASHX_BASE_URI', localProperties.getProperty('DASHX_BASE_URI')
+        buildConfigField 'String', 'DASHX_TARGET_ENVIRONMENT', localProperties.getProperty('DASHX_TARGET_ENVIRONMENT')
```
This seems like too much code. Maybe need to research some best practices around this @kiranattilicho
